new_device.py

```diff
@@ -0,0 +1,171 @@
+import torch
+from torch.overrides import enable_torch_function_mode, TorchFunctionMode
+from torch.utils._pytree import tree_map
+
+import numpy as np
+
+aten = torch.ops.aten
+
+# 1. A Tensor that stores custom raw_data and implement functions for it
+class MyDeviceTensor(torch.Tensor):
+    IMPLEMENTATIONS = {}
+
+    @staticmethod
+    def __new__(cls, size, dtype, raw_data=None, requires_grad=False):
+        # Use a meta Tensor here to be used as the wrapper
+        return torch.Tensor._make_subclass(cls, torch.empty(size, dtype=dtype, device="meta"), require_grad=requires_grad)
```

Comment:
No way to set the device alas!

Reply:
No, all the C++ code that sets "device" will make sure it is one of the ones in our hardcoded list :(